### ISSUE TITLE
Stack overflow using Automate::launch

### ISSUE THREAD

**Issue by ThatsNoMoon** [mbenoukaiss/automate#4]
I'm getting a stack overflow (`thread main has overflowed its stack`) on windows when using the `Automate::launch` convenience method as in the example in the readme. I don't see the same error when I use `#[tokio::main]` myself as in the below:
```rust
#[tokio::main]
async fn main() {
	let config = Configuration::from_env("DISCORD_API_TOKEN")
		.register(stateless!(say_hello));

	ShardManager::with_config(config).await.unwrap().auto_setup().launch().await;
}
```
I can reproduce this on the latest crates.io release and on `master`.

**Comment by mbenoukaiss** [comment]
The only difference there is with using `#[tokio::main]` and manually launching it is that you don't set the `logger::TASK_NAME` for the main task. So I tested removing the code that sets the task name and it's working with `Automate::launch`. 

I don't know why it would cause the stack to overflow only on Windows and not on Linux. It also only overflows in debug mode and not release, so it may be related to the issue mentioned here https://github.com/tokio-rs/tokio/issues/2055. I'll make a commit to fix this issue later this afternoon.

**Comment by mbenoukaiss** [comment]
So I made some changes that fixed it for me on windows in commit 496d857fea7ca0a922026506afb11b5fb9f3283b (removing the `logger::setup_for_task("main", future)`. I invite you to try to use `Automate::launch` on  the latest commit with `automate = { git = "https://github.com/mbenoukaiss/automate", hash = "0015bba" }` to confirm it also works for you.

I also noticed the way the shard manager would start the tasks wasn't really right so I changed that too (it would start the tasks before launch, which works fine but may not be what everyone wants).

And thank you for contributing by reporting these issues!
